batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/BatchProjectFactory.java

```diff
@@ -0,0 +1,78 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.batch.internal.core;
+
+import java.lang.reflect.InvocationTargetException;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+import org.jboss.tools.batch.core.BatchCorePlugin;
+import org.jboss.tools.batch.internal.core.impl.BatchProject;
+import org.jboss.tools.jst.web.kb.KbProjectFactory;
+import org.jboss.tools.jst.web.kb.internal.KbProject;
+
+/**
+ * 
+ * @author Viacheslav Kabanovich
+ *
+ */
+public class BatchProjectFactory {
+	private static final String MODEL_ID = "Batch";
+	
+	public static IBatchProject getBatchProject(IProject project, boolean resolve) {
+		BatchProject result = null;
+		KbProject kb = (KbProject)KbProjectFactory.getKbProject(project, resolve);
+		
+		if(kb != null) {
+			result = (BatchProject)kb.getExtensionModel(MODEL_ID);
+			if(result == null) {
```

Comment:
Shouldn't we synchronize this block somehow to avoid situation when different BatchProject's created and returned?